### PR TITLE
Budget App

### PR DESCRIPTION
**Pull request by AyoubMs**
In this commit, I added:
- devise for user authentication
- routes for each page
- controllers

### CODE REVIEW

**Review by sja-thedude — CHANGES_REQUESTED** [review]
## Required Changes ♻️

Hi @AyoubMs, This is @sja-thedude your code reviewer.

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!

768070f3-bc9a-4aba-b8bc-25793e00ea1e_text

## Highlights
✔️ All linters are passing
✔️ Professional Readme
✔️ Correct git-flow

## Required 
- Kindly make sure to add the video presentation on the loom and add the link in readme and PR:
- [ ] The recorded video presents the features of the project.
- [ ] The maximum length of the video is 5 minutes.
- [ ] The video contains a clear and objective description of the project in professional English. 
- [ ] The topic and purpose of the project are made clear.
- [ ] The video contains a demo of the project, highlighting all the project features.
- [ ] The video contains a highlight about a piece of the code the student finds interesting or is proud of making.
- Deployment:
- [ ] Kindly make sure to deploy your app on Heroku and add the link in the readme.
- Testing:
- [ ] Kindly add unit tests and integration tests for all the most important components of the RoR application.
- Error while Login:
- [ ] When I tried to log in, I encountered an error shown below:
As I couldn't log in, I could not check:
- [ ] Home page and its requirements
- [ ] Transaction page and its requirements
- [ ] Add a new category page and its requirements
- [ ] Add a new transaction page and its requirements
Screenshot 2022-08-02 at 8 28 59 PM

Note: Kindly follow the same font face, size, weight, colors, composition and space bet elements.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@sja-thedude] in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by AnselemOdims — CHANGES_REQUESTED** [review]
### Hi @AyoubMs , 
You have been doing a tremendous job so far :ok_hand:
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
No linter errors :heavy_check_mark:

## Required Changes ♻️
- [ ] As the previous reviewer has pointed out, please ensure that the presentation video is viewed and accessed. At the moment, it only show a black screen like something below
image

- [ ] When I try to signup or login on heroku, I get this error. Please help fix it
image

- [ ]  When I tried to log in, I encountered an error shown above:
          As I couldn't log in, I could not check:
- Home page and its requirements
- Transaction page and its requirements
- Add a new category page and its requirements
- Add a new transaction page and its requirements

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity! You can tag me using @AnselemOdims for a quicker response

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by sja-thedude — CHANGES_REQUESTED** [review]
## Required Changes ♻️

Hi @AyoubMs, This is @sja-thedude your code reviewer.

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

200w

 ## Highlights
✔️ Good job so far

 ## Required Changes ♻️
- [ ] Kindly record another loom video explaining the features of the project, your interesting piece of code and the working of the app.
- [x] Kindly fix the errors for tests shown below:
Screenshot 2022-08-03 at 4 03 38 PM


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@sja-thedude] in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md [hunk 1]
```diff
@@ -1,24 +1,150 @@
...
+Contributions, issues, and feature requests are welcome!
+
+Feel free to check the [issues page](../../issues/).
+
+## Show your support
+
+Give a ⭐️ if you like this project!
+
+## Acknowledgments
+
```
Comment (rloterh) [inline README.md:142]: - [ ] OPTIONAL
It's a nice and smart thing to give appropriate credit to **[Gregoire Vella on Behance](https://www.behance.net/gregoirevella)**, the author of [the original design](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)), as required by the [Creative Commons License](https://creativecommons.org/licenses/). Kindly ensure to add this under the **Acknowledgements** section of your ReadMe.

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Ayoub,  

Your project is complete, Great job! 🥇 There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉🌟 

## Highlights 👍
- App is running as intended

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

_Kindly consider my comments marked **[OPTIONAL]** under this review_
<br>
Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**